pyQuARC/code/datetime_validator.py

```diff
@@ -143,13 +143,38 @@ def validate_datetime_against_granules(
 
         validity = True
         last_granule_datetime = None
-
+        date_time = None
+
+        # Define the formats in decreasing order of precision
+        formats = [
+            "%Y-%m-%dT%H:%M:%S.%f",  # Year to microsecond
+            "%Y-%m-%dT%H:%M:%S",  # Year to second
+            "%Y-%m-%dT%H:%M",  # Year to minute
+            "%Y-%m-%dT%H",  # Year to hour
+            "%Y-%m-%d",  # Year to day
+            "%Y-%m",  # Year to month
+            "%Y",  # Year
+        ]
```

Comment:
Can move this to a constant/config

Reply:
Yes, I can. I moved this to the constants.py file.